### PR TITLE
optional oidc-ca-file flag

### LINKED ISSUE #2467: Make `oidc-ca-file` flag optional
**User Story**
Right now the CA bundle file is mandatory to start controller-manger. This file should be optional and set only if we want to enable TLS verification. 
**Acceptance criteria**
empty or missing `oidc-ca-file` parameter leads to:
- dex-ca secret for user cluster is created only when `oidc-ca-file` flag in controller is specified 
- user cluster API deployment contains `dex-ca` secret, mountPath: `/etc/kubernetes/dex/ca` and `--oidc-ca-file=/etc/kubernetes/dex/ca/caBundle.pem` optionally depends on `oidc-ca-file` flag in controller

### PR DESCRIPTION
**Pull request by kgroschoff**
paired with @zreigz 

**What this PR does / why we need it**:
Right now the CA bundle file is mandatory to start controller-manger. This file should be optional and set only if we want to enable TLS verification.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2467 

**Special notes for your reviewer**:
/

**Release note**:

```release-note
Setting CA bundle with the `oidc-ca-file flag` is now optional
```

### CODE REVIEW

File: api/pkg/controller/cluster/resources_test.go [hunk 1]
```diff
@@ -203,7 +203,7 @@ func TestSecretV2CreatorsKeepAdditionalData(t *testing.T) {
 
 	templateData := resources.NewTemplateData(cluster, dc, "", secretLister, nil, serviceLister, "", "", "", resource.Quantity{}, "", "", false, false, "", nil, file.Name(), "", "")
 
-	for _, op := range GetSecretCreatorOperations(cluster, []byte{}) {
+	for _, op := range GetSecretCreatorOperations(cluster, []byte{}, false) {
```
Comment (p0lyn0mial) [inline api/pkg/controller/cluster/resources_test.go]: could we set it to `true` ? To test `DexCASecret`creation.

**Review by p0lyn0mial — COMMENTED** [review]
I think we also need to update the deployment, https://github.com/kubermatic/kubermatic/blob/master/config/kubermatic/templates/kubermatic-controller-manager-dep.yaml#L66

`-oidc-ca-file` should be set when `OpenIDAuthPlugin` feature was enabled. (https://github.com/kubermatic/kubermatic/blob/master/config/kubermatic/templates/kubermatic-controller-manager-dep.yaml#L67)

**Comment by kgroschoff** [comment]
@p0lyn0mial done 😃

**Comment by zreigz** [comment]
additionally, those parts should be also optional:
https://github.com/kubermatic/kubermatic/blob/5c2df77c1f3a2d68cfe8e1d0062fa21ad03aa60d/config/kubermatic/templates/kubermatic-controller-manager-dep.yaml#L95

https://github.com/kubermatic/kubermatic/blob/5c2df77c1f3a2d68cfe8e1d0062fa21ad03aa60d/config/kubermatic/templates/kubermatic-controller-manager-dep.yaml#L148

**Comment by zreigz** [comment]
Besides of those changes the main functionality works fine

**Comment by zreigz** [comment]
@kgroschoff PTAL

**Comment by kgroschoff** [comment]
You're absolutely right, @zreigz . Added it, could you PTAL again? 😄

File: api/pkg/controller/cluster/resources.go [hunk 2]
```diff
@@ -197,15 +197,15 @@ type SecretOperation struct {
...
+func GetSecretCreatorOperations(c *kubermaticv1.Cluster, dockerPullConfigJSON []byte, enableDexCA bool) []SecretOperation {
 	secrets := []SecretOperation{
 		{resources.CASecretName, certificates.RootCA},
 		{resources.FrontProxyCASecretName, certificates.FrontProxyCA},
 		{resources.ImagePullSecretName, resources.ImagePullSecretCreator(resources.ImagePullSecretName, dockerPullConfigJSON)},
 		{resources.ApiserverFrontProxyClientCertificateSecretName, apiserver.FrontProxyClientCertificate},
 		{resources.EtcdTLSCertificateSecretName, etcd.TLSCertificate},
 		{resources.ApiserverEtcdClientCertificateSecretName, apiserver.EtcdClientCertificate},
-		{resources.DexCASecretName, apiserver.DexCACertificate},
+
```
Comment (zreigz) [inline api/pkg/controller/cluster/resources.go]: Please remove the empty line

**Review by zreigz — CHANGES_REQUESTED** [review]

**Comment by zreigz** [comment]
Maybe it would be good also try to rebase this code

**Comment by kgroschoff** [comment]
@zreigz done 👍

**Comment by zreigz** [comment]
/lgtm
@p0lyn0mial , please, look again, if we have not forgotten something

**Comment by p0lyn0mial** [comment]
thank you @kgroschoff and @zreigz for implementing and reviewing.

/lgtm
/approve

**Pull request merged**